voxblox/src/integrator/esdf_integrator.cc

```diff
@@ -119,6 +119,16 @@ void EsdfIntegrator::updateFromTsdfLayer(bool clear_updated_flag) {
       }
     }
   }
+
+  // Clear all esdf blocks which are no longer allocated in the corresponding
+  // tsdf layer.
+  BlockIndexList allocated_esdf_blocks;
+  esdf_layer_->getAllAllocatedBlocks(&allocated_esdf_blocks);
+  for (const BlockIndex& esdf_block_index : allocated_esdf_blocks) {
+    if (!tsdf_layer_->hasBlock(esdf_block_index)) {
+      esdf_layer_->removeBlock(esdf_block_index);
```

Comment:
Don't know voxblox well enough - I assume the `esdf_block_index` are global, so removing one block does not shift the indices of the remaining ones? (Otherwise, the `removeBlock` in the loop could remove the wrong blocks)

Reply:
yes, exactly, the block indices are global